src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -328,6 +375,24 @@ bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(app::CommandH
     return true;
 }
 
+bool emberAfGeneralCommissioningClusterSetTCAcknowledgementsCallback(
+    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::GeneralCommissioning::Commands::SetTCAcknowledgements::DecodableType & commandData)
+{
+#if CHIP_CONFIG_TC_REQUIRED
+    MATTER_TRACE_SCOPE("SetTCAcknowledgements", "GeneralCommissioning");
+    Commands::SetTCAcknowledgementsResponse::Type response;
+    EnhancedSetupFlowProvider * const enhancedSetupFlowProvider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+    uint16_t acknowledgements                                   = commandData.TCUserResponse;
+    uint16_t acknowledgementsVersion                            = commandData.TCVersion;
+    CheckSuccess(enhancedSetupFlowProvider->SetTermsAndConditionsAcceptance(acknowledgements, acknowledgementsVersion), Failure);
+    response.errorCode = CommissioningErrorEnum::kOk;
+
+    commandObj->AddResponse(commandPath, response);
+#endif
+    return true;
```

Comment:
are we supposed to have this lined in even if TC_REQUIRED is not there? I see no AddResponse here, so will this make the commandObj dangling ? does commandObj expect that AddResponse must be called? If so even the non-enabled path has to have this AddResponse or we should condition-out this entire callback.

Reply:
I'm not sure what to do here. If the command isn't supported, there isn't an appropriate response code. I'm updating to return  false to indicate that the handling failed.